### ISSUE TITLE
Marten LinqExtensions that don't have in memory implementation return true instead of throw

### ISSUE THREAD

**Issue by karelg** [JasperFx/marten#1583]
Marten has some special extensions, that it will recognize when generating SQL, but often there is no in memory implementation for these and they will return just `true`. Unfortunately, this behavior will introduce nasty bug's, when you try to run the same linq query on in memory collections. Preferred behavior for these extensions would be throwing NotSupportedException.

Example:
````
  public static bool Search<T>(this T variable, string searchTerm, string regConfig)
  {
       return true;  // Should throw instead
  }
````

The same thing can happen with MatchesSqlExtensions.

**Comment by karelg** [comment]
Maybe this is something to consider for v4 if it is a too risky change for v3?

**Comment by jeremydmiller** [comment]
@karelg I can't think of any reason why this would be risky for the v3 codebase, and it should be easy to cherry pick the changes to v4. Would you be up for the PR on that?

**Comment by karelg** [comment]
@jeremydmiller Sorry for the late response, didn't notice the reply before. Yes, I can make a PR to address this issue.

**Comment by karelg** [comment]
#1736

**Comment by mysticmind** [comment]
This issue is addressed in PR #1736
